week3/yeonwook/todo/app.js

```diff
@@ -0,0 +1,101 @@
+const todoListEl = document.getElementById("todoList");
+const todoInputEl = document.getElementById("todoInput");
+
+const API_URL = "http://localhost:3001/todos";
+
+fetch(API_URL)
+  .then((response) => response.json())
+  .then((data) => renderTodo(data));
+
+const updateTodo = (todoId, originalTitle) => {
+
+  const updateContainer = document.createElement("li");
+
+  const inputToDoEl = document.createElement("input");
+  const updateButton = document.createElement("button");
+  updateButton.classList.add("updatebutton");
+  updateButton.textContent = "수정완료";
+
+  inputToDoEl.setAttribute("id", "todoInput");
+  inputToDoEl.type = "text";
+  inputToDoEl.placeholder = `${originalTitle} 수정`;
+  updateContainer.append(inputToDoEl);
+  updateContainer.append(updateButton);
+  todoListEl.append(updateContainer)
+
+  updateButton.onclick = ()=>{
+    const newTitle = inputToDoEl.value.toString();
+    if(!newTitle) return;
+    fetch(API_URL + "/" + todoId, {
+      method: "PATCH",
+      headers:{"Content-Type": "application/json"},
+      body : JSON.stringify({
+        title: newTitle, completed : false
+      })
+    })
+    .then((response) => response.json())
+    .then((data) => renderTodo(data));
+    todoListEl.remove(inputToDoEl);
+    todoListEl.remove(updateButton);
+  }
+
+};
+
+const renderTodo = (newTodos) => {
+  todoListEl.innerHTML = "";
+  newTodos.forEach((todo) => {
+    const listEl = document.createElement("li");
+    listEl.textContent = todo.title;
+    listEl.id = `todo-${todo.id}`;
+
+    const deleteEl = document.createElement("button");
+    deleteEl.textContent = "🗑️";
+    deleteEl.onclick = () => deleteTodo(todo.id);
+
+    const udpateEl = document.createElement("button");
+    udpateEl.textContent = "✏️";
+    udpateEl.onclick = () => updateTodo(todo.id, todo.title);
+
+    listEl.append(deleteEl);
+    listEl.append(udpateEl);
+    todoListEl.append(listEl);
+  });
+};
+
+const addTodo = () => {
+  const title = todoInputEl.value;
+  const date = new Date();
+  const createdAt = date.toDateString();
+
+  if (!title) return;
+
+  const newTodo = {
+    id: date.getTime(),
```

Comment:
date.getTime()은 number를 반환하는 함수이기 때문에 .toString()을 붙여주시면 좋을 것 같습니다!
db.json에 있는 id는 다 string이어서 타입을 맞춰주시는 것이 좋습니다. 
```suggestion
    id: date.getTime().toString(),
```